stock_move_actual_date/tests/test_stock_move_actual_date.py

```diff
@@ -0,0 +1,147 @@
+                "type": "product",
+                "categ_id": product_category.id,
+                "standard_price": 100.0,
+            }
+        )
+        cls.product_2 = cls.env["product.product"].create(
+            {
+                "name": "Test Product 2",
+                "type": "product",
+                "categ_id": product_category.id,
+                "standard_price": 0.0,
+            }
+        )
+        cls.supplier_location = cls.env.ref("stock.stock_location_suppliers")
+        cls.stock_location = cls.env.ref("stock.stock_location_stock")
+
+    def create_picking(self, actual_date=False):
+        receipt = self.env["stock.picking"].create(
+            {
+                "location_id": self.supplier_location.id,
+                "location_dest_id": self.stock_location.id,
+                "partner_id": self.partner.id,
+                "picking_type_id": self.env.ref("stock.picking_type_in").id,
+                "actual_date": actual_date,
+            }
+        )
+
+        move = self.env["stock.move"].create(
+            {
+                "picking_id": receipt.id,
+                "name": "10 in",
+                "location_id": self.supplier_location.id,
+                "location_dest_id": self.stock_location.id,
+                "product_id": self.product_1.id,
+                "product_uom_qty": 10.0,
+                "price_unit": 10,
+                "move_line_ids": [
+                    Command.create(
+                        {
+                            "product_id": self.product_1.id,
+                            "location_id": self.supplier_location.id,
+                            "location_dest_id": self.stock_location.id,
+                            "qty_done": 10.0,
+                        }
+                    )
+                ],
+            }
+        )
+        move._action_confirm()
+        move._action_done()
+        return receipt, move
+
+    def create_scrap(self, receipt, actual_date=False):
+        scrap = self.env["stock.scrap"].create(
+            {
+                "product_id": self.product_1.id,
+                "scrap_qty": 2.0,
+                "picking_id": receipt.id,
+                "actual_date": actual_date,
+            }
+        )
+        scrap.action_validate()
+        return scrap
+
+    def test_stock_move_actual_date(self):
+        receipt, move = self.create_picking(date(2024, 9, 1))
+        self.assertEqual(move.actual_date, date(2024, 9, 1))
+        self.assertEqual(move.account_move_ids.date, date(2024, 9, 1))
+        receipt.actual_date = date(2024, 8, 1)
+        self.assertEqual(move.actual_date, date(2024, 8, 1))
+        self.assertEqual(move.account_move_ids.date, date(2024, 8, 1))
+        scrap = self.create_scrap(receipt, date(2024, 9, 10))
+        self.assertEqual(scrap.move_id.actual_date, date(2024, 9, 10))
+        self.assertEqual(scrap.move_id.account_move_ids[0].date, date(2024, 9, 10))
```

Comment:
@AungKoKoLin1997 I think we should be able to remove the temp adjustment now. Please update all the relevant open PRs.
```suggestion
        self.assertEqual(scrap.move_id.account_move_ids.date, date(2024, 9, 10))
```